usr/share/rear/restore/BORG/default/300_load_archives.sh

```diff
@@ -17,17 +17,47 @@ fi
 # Display header.
 LogUserOutput "
 === Borg archives list ===
-Location:   ${BORGBACKUP_HOST:-USB}
-Repository: $BORGBACKUP_REPO
-"
+
+Location:           ${BORGBACKUP_HOST:-USB}
+Repository:         $BORGBACKUP_REPO
+Number of archives: $archive_cache_lines"
 
 # Display BORGBACKUP_ARCHIVE_CACHE file content
 # and prompt user for archive to restore.
 # Always ask which archive to restore (even if there is only one).
 # This gives possibility to abort restore if repository doesn't contain
 # desired archive, hence saves some time.
+
+# Pagination for selecting archives:
+# Show BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER archives at a time, starting
+# with the current ones.
+# If no valid choice is given, cycle through older archives.
+# Enabled by default (BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=10).
+# To disable pagination set BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=0.
+
+archive_cache_last_shown=0
+
+# For timestamp output of Borg archives ISO 8601 format is used:
+# YYYY-MM-DDThh:mm:ss, e.g.: 2020-05-26T00:25:00
+
+# When pagination is disabled by the user, show everything
+[[ $BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER -eq 0 ]] \
+    && BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER=$archive_cache_lines
+
 while true ; do
-    LogUserOutput "$( cat -n "$BORGBACKUP_ARCHIVE_CACHE" | awk '{print "["$1"]", $2,"\t"$3,$4,$5}' )"
+    UserOutput ""
+    LogUserOutput "$( cat -n "$BORGBACKUP_ARCHIVE_CACHE" \
+        | awk '{print "["$1"]", $4 "T" $5, $2 }' \
+        | head -n $(( archive_cache_lines - archive_cache_last_shown )) \
+        | tail -n "$BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER" )"
+    (( archive_cache_last_shown += BORGBACKUP_RESTORE_ARCHIVES_SHOW_NUMBER ))
```

Comment:
Thank you so much to have spaces in the code for better readability!
It is much appreciated - at least my eyes do much appreciate it ;-)

Reply:
Thanks for your review and pointing out several things.